hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -62,4 +63,7 @@
      * @return sampled entries.
      */
     Iterable<LazyEntryViewFromRecord> getRandomSamples(int sampleCount);
+
+    MapKeyIteratorResult fetchNext(int tableIndex, int size);
+
```

Comment:
You are right naming is not clear. how about `fetchKeysFromIndex()` and `MapPartialFetchResult` ?